hs_core/views/resource_rest_api.py

```diff
@@ -896,6 +908,10 @@ def post(self, request, pk):
     def delete(self, request, pk, filename):
         resource, _, user = view_utils.authorize(
             request, pk, needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+        view_utils.irods_path_is_allowed(filename)
+        if not resource.supports_folders and '/' in filename:
+            raise ValidationError("Resource type does not support folders")
+
```

Comment:
@alvacouch It is better to move ```view_utils.irods_path_is_allowed(filename)``` after line 914 and only call it when ```'/' in filename``` is true since in majority of cases, users may just want to add file to the root folder in which case path check is not needed.